client/src/api/document.ts

```diff
@@ -43,6 +44,7 @@ export namespace Document {
         switch (res.status) {
             case StatusCodes.CREATED: return SnapshotSchema.shape.creation.parse(await res.json());
             case StatusCodes.CONFLICT: return BarcodeAssignmentErrorSchema.parse(await res.json());
+            case StatusCodes.SERVICE_UNAVAILABLE: throw new DeferredSnap;

```

Comment:
Stylistically, I would prefer to put this below the `4xx` errors. It's not a deal-breaker, but I'm nitpicking the code consistency here.